clash-lib/src/Clash/Core/Util.hs

```diff
@@ -479,6 +478,12 @@ undefinedPrims =
   , "GHC.Real.underflowError"
   ]
 
+undefinedXPrims :: [T.Text]
+undefinedXPrims =
+  [ "Clash.Normalize.Primitives.undefinedX"
+  , "Clash.XException.errorX"
```

Comment:
Something that just occurs to me: since `errorX` is just `throw (XException ...)` it presumably always gets left alone by the simplifier, but what if a user writes `throw (XException "Boom")` directly? Does this then fail to be identified as an _undefined X_ primitive and get identified as a normal undefined primitive?

Reply:
There's no eval rule for `throw` right now, so the whole evaluator will reduce to `Nothing`. I'll add a new issue so this doesn't get lost.

Reply:
https://github.com/clash-lang/clash-compiler/issues/2158